src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx

```diff
@@ -237,4 +237,17 @@ class MyEventWatcher
 end
 ```
 
-For more information, see [Tracer#start_transaction_or_segment in the New Relic RubyDoc](https://www.rubydoc.info/github/newrelic/newrelic-ruby-agent/NewRelic/Agent/Tracer#start_transaction_or_segment-class_method).
+For more information, see [Tracer#start_transaction_or_segment in the New Relic RubyDoc](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/Agent/Tracer#start_transaction_or_segment-class_method).
+
+### Instrumenting Threads
+
+To support applications that want to control which threads are instrumented by New Relic, a class is available in 8.7.0 and above, `NewRelic::TracedThread`, that will create a thread that includes New Relic instrumentation. 
+Use this class as a replacement for the native Thread class. Please see our [API documentation](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/TracedThread) for more details. 
+```
+  NewRelic::TracedThread.new do 
+    # custom code to run inside of the thread
+  end
+
+```
+To have the agent automatically instrument all threads, please see the configuration `instrumentation.thread.tracing` [documentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#instrumentation).
```

Comment:
```suggestion
In 8.7.0 and higher, use the class `NewRelic::TracedThread` in applications to create threads that are instrumented by New Relic.

Use this class as a replacement for the native `Thread` class. See our [TracedThread API documentation](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/TracedThread) for more detail. 

```
  NewRelic::TracedThread.new do 
    # custom code to run inside of the thread
  end

```

To automatically instrument all threads, see the [configuration `instrumentation.thread.tracing` documentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#instrumentation).
```